db/db_iter.cc

```diff
@@ -933,12 +933,7 @@ void DBIter::FindPrevUserKey() {
     return;
   }
   size_t num_skipped = 0;
-// Suppress false positive clang analyzer warnings.
-#ifdef __clang_analyzer__
-  ParsedInternalKey ikey(Slice(), 0, 0);
```

Comment:
Sure, I definitely can, but was wondering why we need to do a different thing for clang_analyzer. 
There are definitely some false positive that show up with clang_analayzer and its not perfect. I don't don't know of a precedent here in our code base, hence was wondering.
Btw, I also filed some bugs against clang analyzer yesterday, as it was also complaining about stl files, but they might be fixed in the latest versions of the analyzer.